opencti-platform/opencti-front/src/private/components/settings/ThemesEditor.tsx

```diff
@@ -0,0 +1,488 @@
+import { Button, Paper } from '@mui/material';
+import { useTheme } from '@mui/styles';
+import React, { FormEvent, FunctionComponent, useState } from 'react';
+import { Field, Form, Formik } from 'formik';
+import { Disposable, graphql, useLazyLoadQuery } from 'react-relay';
+import * as Yup from 'yup';
+import { useFormatter } from '../../../components/i18n';
+import AutocompleteField from '../../../components/AutocompleteField';
+import ColorPickerField from '../../../components/ColorPickerField';
+import { SubscriptionFocus } from '../../../components/Subscription';
+import TextField from '../../../components/TextField';
+import { commitMutation, defaultCommitMutation, MESSAGING$ } from '../../../relay/environment';
+import type { Theme } from '../../../components/Theme';
+import ThemeCreator, { createThemeMutation, CustomThemeBaseType } from './ThemeCreator';
+import { ThemesEditor_themes$data } from './__generated__/ThemesEditor_themes.graphql';
+import { ThemesEditorAboutQuery } from './__generated__/ThemesEditorAboutQuery.graphql';
+import VisuallyHiddenInput from '../common/VisuallyHiddenInput';
+import useApiMutation from '../../../utils/hooks/useApiMutation';
+import { ThemeCreatorCreateMutation } from './__generated__/ThemeCreatorCreateMutation.graphql';
+
+export const refetchableThemesQuery = graphql`
+  fragment ThemesEditor_themes on Query
+  @refetchable(queryName: "ThemesEditorRefetchQuery") {
+    themes {
+      edges {
+        node {
+          id
+          name
+          theme_background
+          theme_paper
+          theme_nav
+          theme_primary
+          theme_secondary
+          theme_accent
+          theme_logo
+          theme_logo_collapsed
+          theme_logo_login
+        }
+      }
+    }
+  }
+`;
+
+const editThemeMutation = graphql`
+  mutation ThemesEditorEditMutation(
+    $id: ID!,
+    $input: [EditInput!]!,
+  ) {
+    themeFieldPatch (id: $id, input: $input) {
+      id
+      name
+      theme_background
+      theme_paper
+      theme_nav
+      theme_primary
+      theme_secondary
+      theme_accent
+      theme_logo
+      theme_logo_collapsed
+      theme_logo_login
+    }
+  }
+`;
+
+const aboutQuery = graphql`
+  query ThemesEditorAboutQuery {
+    about {
+      version
+    }
+  }
+`;
+
+export interface ThemeType extends CustomThemeBaseType {
+  id: string;
+}
+
+interface ThemesEditorProps {
+  themes: ThemesEditor_themes$data['themes'];
+  refetch: () => Disposable;
+  editContext?: {
+    name: string,
+    focusOn?: string,
+  }[];
+  currentTheme?: string | null;
+}
+
+const ThemesEditor: FunctionComponent<ThemesEditorProps> = ({
+  themes,
+  refetch,
+  editContext,
+  currentTheme,
+}) => {
+  const theme = useTheme<Theme>();
+  const { t_i18n } = useFormatter();
+  const [open, setOpen] = useState<boolean>(false);
+  const { about } = useLazyLoadQuery<ThemesEditorAboutQuery>(aboutQuery, {});
```

Comment:
We prefer to use `usePreloadedQuery` instead of `useLazyLoadQuery`.
=> First you need to prepare the query Ref in a parent componant using useQueryLoading and then replace useLazyLoadQuery with usePreloadedQuery